lang/mrustc/Portfile

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem      1.0
+PortGroup       github 1.0
+
+github.setup    thepowersgang mrustc aa4adf3504876f58c90a3b8770116fbefc7c2867
+
+version         20220103
+revision        0
+categories      lang devel
+license         MIT
+maintainers     nomaintainer
+description     Alternative implementation of Rust
+supported_archs x86_64
+
+long_description \
+                In-progress alternative rust compiler. Capable of building a \
+                fully-working copy of rustc, but not suitable for everyday use \
+                (due to inadequate error messages).
+
+master_sites-append   https://static.rust-lang.org/dist/
+
+set rust_version 1.39.0
+set rust_version_major [join [lrange [split ${rust_version} .-] 0 1] .]
+set rust_version_major_underscore [join [lrange [split ${rust_version} .-] 0 1] _]
+distfiles-append rustc-${rust_version}-src.tar.gz
+
+checksums       mrustc-${github.version}.tar.gz \
+                rmd160  1e76cd4451a096629c855c6786bbf5c93233872d \
+                sha256  4b58ffe5315081ea780512c8930875ae7d3fc41adfc4d7554c0738af50cbaaf6 \
+                size    1171492 \
+                rustc-${rust_version}-src.tar.gz \
+                rmd160  d5b04b87fc336e3be7d592f70de0363aa66622aa \
+                sha256  b4a1f6b6a93931f270691aba4fc85eee032fecda973e6b9c774cd06857609357 \
+                size    152803201
+
+use_configure   no
+
+compiler.cxx_standard 2014
+compiler.c_standard 2011
+
+# This will be the default compiler, which can be overridden
+# at run-time with the CC environment variable.
+# Support variants in the future, but this will do for now
+depends_run-append port:clang-9.0
```

Comment:
It would be better to not hardcode a version here, but to tweak the compiler backlisting etc. to ensure a compatible compiler is picked, and then to depend on that. This would allow newer compilers to be used when, e.g., clang-9.0 is not available (such as on newer systems which inevitably when one day stop supporting this compiler).

Note to add the correct runtime dep you can use the same logic as in rust.

https://github.com/macports/macports-ports/blob/a34b9f304f3a13bb9239e2ae9917ee3a41de3e9f/lang/rust/Portfile#L210